src/validation.cpp

```diff
@@ -3253,8 +3263,6 @@ static bool AcceptBlock(const CBlock& block, CValidationState& state, CBlockInde
     int nHeight = pindex->nHeight;
 
     if (isPoS) {
-        LOCK(cs_main);
-
```

Comment:
Why this lock removal?

The `IsUnspentOnFork` function that is below accesses to `chainActive`, the `Reject forks below max depth` validation and `IsSpentOnActiveChain` too.

Reply:
The caller of `AcceptBlock` already has locked cs_main, and we have asserted it. No need to re-lock here.